From 6decafbdf2f2f7df8302c41f060b2c5966722ed1 Mon Sep 17 00:00:00 2001
From: Richard Feliciano <rfeliciano@BeaconEmbedded.com>
Date: Tue, 26 Dec 2023 19:52:57 -0600
Subject: [PATCH] Remove the ignore regulatory command from wifi driver

The REGULATORY_IGNORE_STALE_KICKOFF set by NXP is not recognized
by the updated kernel.  In other drivers the flag is conditial
based on kernel revs.  So the code was modified as other drivers
using this flag

Signed-off-by: Richard Feliciano <rfeliciano@BeaconEmbedded.com>

diff --git a/mxm_wifiex/wlan_src/mlinux/moal_sta_cfg80211.c b/mxm_wifiex/wlan_src/mlinux/moal_sta_cfg80211.c
index f13f641..4690a3b 100644
--- a/mxm_wifiex/wlan_src/mlinux/moal_sta_cfg80211.c
+++ b/mxm_wifiex/wlan_src/mlinux/moal_sta_cfg80211.c
@@ -10135,12 +10135,14 @@ mlan_status woal_register_cfg80211(moal_private *priv)
 	}
 #endif
 #if CFG80211_VERSION_CODE >= KERNEL_VERSION(3, 19, 0)
+#if CFG80211_VERSION_CODE < KERNEL_VERSION(6, 1, 0)
 	/*REGULATORY_IGNORE_STALE_KICKOFF: the regulatory core will _not_ make
 	 * sure all interfaces on this wiphy reside on allowed channels. If this
 	 * flag is not set, upon a regdomain change, the interfaces are given a
 	 * grace period (currently 60 seconds) to disconnect or move to an
 	 * allowed channel.*/
 	wiphy->regulatory_flags |= REGULATORY_IGNORE_STALE_KICKOFF;
+#endif
 #endif
 
 	memset(&priv->phandle->country_code, 0,
-- 
2.34.1

